contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,864 @@
+	 * @param from The address from which the borrow has to be repaid on behalf of 'to'.
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param amt The amount to be repaid.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackFromUsingManager(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+		require(
+			market != address(0) && token != address(0) && to != address(0),
+			"invalid market/token/to address"
+		);
+
+		address token_ = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		require(token == token_ || isEth, "invalid-token");
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		amt_ = _calculateFromAmount(
+			market,
+			token_,
+			from,
+			amt_,
+			isEth,
+			Action.REPAY
+		);
+
+		uint256 supplyBalance_ = CometInterface(market).balanceOf(to);
+		require(supplyBalance_ == 0, "cannot-repay-when-supplied");
+
+		CometInterface(market).supplyFrom(from, to, token_, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPaybackFromUsingManager(address,address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token_, from, to, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Buy collateral asset absorbed, from the market.
+	 * @notice Buy collateral asset to increase protocol base reserves until targetReserves is reached.
+	 * @param market The address of the market from where to withdraw.
+	 * @param asset The collateral asset to purachase.
+	 * @param dest The address to transfer the purchased assets.
+	 * @param minCollateralAmt Minimum amount of collateral expected to be received.
+	 * @param baseAmt Amount of base asset to be sold for collateral.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of base tokens sold.
+	 */
+	function buyCollateral(
+		address market,
+		address asset,
+		address dest,
+		uint256 minCollateralAmt,
+		uint256 baseAmt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, baseAmt);
+		require(
+			market != address(0) && asset != address(0) && dest != address(0),
+			"invalid market/token/to address"
+		);
+
+		bool isEth = asset == ethAddr;
+		address token_ = isEth ? wethAddr : asset;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		convertEthToWeth(isEth, tokenContract, amt_);
+		approve(TokenInterface(getBaseToken(market)), market, amt_);
+
+		CometInterface(market).buyCollateral(
+			asset,
+			minCollateralAmt,
+			amt_,
+			dest
+		);
+
+		uint256 collAmt = CometInterface(market).quoteCollateral(asset, amt_);
```

Comment:
@Richa-iitr Why are you calling the quoteCollateral after buyCollateral?

Reply:
for setting the setId